app/views/admin/editions/_nation_fields.html.erb

```diff
@@ -0,0 +1,75 @@
+<%= hidden_field_tag "edition[nation_inapplicabilities_attributes][0][id]", edition.nation_inapplicabilities.find_by(nation_id: 1)&.id %>
+<%= hidden_field_tag "edition[nation_inapplicabilities_attributes][1][id]", edition.nation_inapplicabilities.find_by(nation_id: 1)&.id %>
+<%= hidden_field_tag "edition[nation_inapplicabilities_attributes][2][id]", edition.nation_inapplicabilities.find_by(nation_id: 1)&.id %>
+<%= hidden_field_tag "edition[nation_inapplicabilities_attributes][3][id]", edition.nation_inapplicabilities.find_by(nation_id: 1)&.id %>
+<%= hidden_field_tag "edition[nation_inapplicabilities_attributes][0][nation_id]", 1 %>
+<%= hidden_field_tag "edition[nation_inapplicabilities_attributes][1][nation_id]", 2 %>
+<%= hidden_field_tag "edition[nation_inapplicabilities_attributes][2][nation_id]", 3 %>
+<%= hidden_field_tag "edition[nation_inapplicabilities_attributes][3][nation_id]", 4 %>
```

Comment:
This (and the controller munging) feels a bit like we're bending over backwards to get the [checkboxes component][] to fit our needs.

I find these `hidden_field_tags` a bit confusing – I'm not entirely sure what they're doing or why. But I suspect there could be a bug hiding in here: the first four lines all perform a `find_by(nation_id: 1)` – hardcoded to `nation_id: 1`, whereas I think they might need to be for IDs 1, 2, 3 and 4 as the following four lines are?

I wonder if a nicer approach might be to break away from using the publishing component in this case and go with custom HTML markup. That way we'd (hopefully) be able to do away with the data munging in the controller.

What do you think?

[checkboxes component]: https://components.publishing.service.gov.uk/component-guide/checkboxes

Reply:
Either way this should probs live in a view component. At a first glance. Does this handle passing a value when something is unchecked?

Reply:
Maybe we can chat this through after standup

Reply:
> Does this handle passing a value when something is unchecked?

No 😞 plus it doesn't set the `nation_id` field for each item, either. So... actually those hidden inputs would still be needed most likely.

Let's chat after standup.

Reply:
After discussion we're going to stick as is for now and i'm going to whack it in a component after